### PR TITLE
Add config for CAPV image promotion

### PR DESCRIPTION
**Pull request by sbueringer**
Part of https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/2051#issuecomment-1653180046

### CODE REVIEW

**Comment by sbueringer** [comment]
cc @randomvariable @srm09 @yastij 
for lgtm from CAPV side

File: registry.k8s.io/manifests/k8s-staging-cluster-api-vsphere/promoter-manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# google group for gcr.io/k8s-staging-cluster-api-vsphere is k8s-infra-staging-cluster-api-vsphere@kubernetes.io
+registries:
```
Comment (sbueringer) [inline registry.k8s.io/manifests/k8s-staging-cluster-api-vsphere/promoter-manifest.yaml:2]: @ameukam Pretty sure the changes in the other files are correct

Do I need this change?

Comment (sbueringer) [reply registry.k8s.io/manifests/k8s-staging-cluster-api-vsphere/promoter-manifest.yaml:2]: Just found the doc. looks like it is correct

File: registry.k8s.io/images/k8s-staging-cluster-api-vsphere/images.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+- name: cluster-api-vsphere-controller
+  dmap:
+    "sha256:TODO": ["v1.8.0-beta.1"]
```
Comment (sbueringer) [inline registry.k8s.io/images/k8s-staging-cluster-api-vsphere/images.yaml:3]: Will update as soon as we have this image

**Comment by sbueringer** [comment]
@ameukam Can you please already take a look and check if this seems right to you?

Context is that we want to introduce image-promotion for CAPV

File: registry.k8s.io/images/k8s-staging-cluster-api-vsphere/images.yaml (hunk 2 above)
Comment (ameukam) [inline registry.k8s.io/images/k8s-staging-cluster-api-vsphere/images.yaml:3]:
```suggestion
# No images yet
```

You don't have wait an image to be built to complete the onboarding.

File: registry.k8s.io/images/k8s-staging-cluster-api-vsphere/OWNERS [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+approvers:
+- chrischdi
+- gab-satchi
+- killianmuldoon
+- randomvariable
+- sbueringer
+- srm09
+- yastij
+
+emeritus_approvers:
```
Comment (ameukam) [inline registry.k8s.io/images/k8s-staging-cluster-api-vsphere/OWNERS:12]: Any reason why this list is the same as the approvers ?

Comment (sbueringer) [reply registry.k8s.io/images/k8s-staging-cluster-api-vsphere/OWNERS:12]: Copy & pasted and thought one of those sections is reviewers. Thx!

**Pull request merged**